Remove --build-at-head flag from anago and releaselib

saschagrunert opened the pull request:
#### What type of PR is this?

/kind api-change

#### What this PR does / why we need it:
The is not actively used any more and removing it will reduce the
technical depth as well as making the transformation to golang easier.
#### Which issue(s) this PR fixes:

<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
-->
None

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->

```release-note
- Removed `--build-at-head` flag from anago
```

tpepper commented:
One small change request:  For the git history which could be viewed disconnected relative to our GitHub based handbooks and build issues / issue template, could the commit message indicate that current preferred invocation is the explicit build version, eg: specifying something like `--buildversion=v1.19.0-rc.2.125+182a67fa7bef52`.

saschagrunert commented:
> One small change request: For the git history which could be viewed disconnected relative to our GitHub based handbooks and build issues / issue template, could the commit message indicate that current preferred invocation is the explicit build version, eg: specifying something like `--buildversion=v1.19.0-rc.2.125+182a67fa7bef52`.

Yep sure, I modified the commit message and rebased on top of the latest master commit.

justaugustus commented:
/lgtm
/approve
/retest
